core/src/main/java/hudson/model/DownloadService.java

```diff
@@ -185,10 +202,27 @@ public static String loadJSON(URL src) throws IOException {
     @Restricted(NoExternalUse.class)
     public static String loadJSONHTML(URL src) throws IOException {
         URLConnection con = ProxyConfiguration.open(src);
+        // JENKINS-47819
+        // A redirection from http to https (or vise versa) returns a 302 response status. Force redirection
```

Comment:
I suspect this is wrong. When using browser-based download, we cannot switch protocols. Really we should delete that option entirely.

Reply:
I tested all these changes in both situations, enabling and disabling browser-based download, and all of them succeeded. As I could see, in browser-based download, all connection are open by browser and this DownloadService is only invoked in Initial Status by SetUpWizard.
However, there could be an use case I don't know, so if you give me an example of when you think it will fail, I could tested.